### ISSUE TITLE
A first proper start up of local server throws some ES errors

### ISSUE THREAD

**Issue by mikelittle** [humanmade/altis-local-server#633]
After an Altis local server stack starts for the first time (after the first failure mentioned in [issue 632](https://app.zenhub.com/workspaces/vulcan-6284e0dfb386b00c83981426/issues/gh/humanmade/altis-local-server/632)), ElasticSearch always throws some errors

```shell
$ composer server start
Starting...
[+] Building 0.0s (0/0)                     docker:colima
[+] Running 1/0
 ✔ Container altis-proxy            Running  0.0s
[+] Running 2/2
 ⠿ Container altis17beta-mutagen-1  Stopped  0.3s
 ⠿ Mutagen                          Paused   0.0s
[+] Running 2/2
 ⠿ Container altis17beta-mutagen-1  Healthy  1.0s
 ⠿ Mutagen                          Started  0.0s
[+] Running 12/12
 ⠿ Container altis17beta-tachyon    Running  0.0s
 ⠿ Container altis17beta-redis      Running  0.0s
 ⠿ Container altis17beta-mailhog    Running  0.0s
 ⠿ Container altis17beta-xray       Running  0.0s
 ⠿ Container altis17beta-s3         Running  0.0s
 ⠿ Container altis17beta-db         Healthy  1.1s
 ⠿ Container altis17beta-s3-sync    Running  0.0s
 ⠿ Container altis17beta-es         Healthy 11.1s
 ⠿ Container altis17beta-php        Started  1.3s
 ⠿ Container altis17beta-cavalcade  Started  1.3s
 ⠿ Container altis17beta-kibana     Started  1.4s
 ⠿ Container altis17beta-nginx      Started  0.8s
Warning: Error in ElasticSearch request:  (404) in /usr/src/app/vendor/altis/cloud/inc/namespace.php on line 389
Warning: Error in ElasticPress request: index_exists  (404) in /usr/src/app/vendor/altis/enhanced-search/inc/namespace.php on line 391
Warning: Error in ElasticSearch request: {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-post-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-post-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-post-2"}],"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-post-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-post-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-post-2"},"status":404} (404) in /usr/src/app/vendor/altis/cloud/inc/namespace.php on line 389
Warning: Error in ElasticPress request: delete_index {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-post-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-post-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-post-2"}],"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-post-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-post-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-post-2"},"status":404} (404) in /usr/src/app/vendor/altis/enhanced-search/inc/namespace.php on line 391
Warning: Error in ElasticSearch request: {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-term-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-term-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-term-2"}],"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-term-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-term-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-term-2"},"status":404} (404) in /usr/src/app/vendor/altis/cloud/inc/namespace.php on line 389
Warning: Error in ElasticPress request: delete_index {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-term-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-term-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-term-2"}],"type":"index_not_found_exception","reason":"no such index [ep-altis17betaaltisdevrepo-term-2]","resource.type":"index_or_alias","resource.id":"ep-altis17betaaltisdevrepo-term-2","index_uuid":"_na_","index":"ep-altis17betaaltisdevrepo-term-2"},"status":404} (404) in /usr/src/app/vendor/altis/enhanced-search/inc/namespace.php on line 391
Setting up ElasticPress...

ElasticPress configured.
Installed database.
WP Username:	admin
WP Password:	password
Startup completed.
To access your site visit: https://altis17beta.altis.dev/
```

This doesn't happen on subsequent starts.

### Acceptance Criteria

- [ ] The errors do not occur

---

### Ready for Work Checklist

Is this ticket ready to be worked on? See
[the Play Book Definition of Ready](https://playbook.hmn.md/play/product/definition-of-ready/)

- [ ] Is the title clear?
- [ ] Is the description clear and detailed enough?
- [ ] Are acceptance criteria listed?
- [ ] Have any (potential) dependencies been identified?
- [ ] Is an estimate or time box assigned?
- [ ] Is a priority label assigned?
- [ ] Is this ticket added to an epic? (Optional)

### Completion Checklist

Is this ticket done? See
[the Play Book Definition of Ready](https://playbook.hmn.md/play/product/definition-of-ready/)

- [ ] Has the acceptance criteria been met?
- [ ] Is the documentation updated (including README)?
- [ ] Does any code/documentation meet project standards?
- [ ] Are automatic tests in place to verify the fix or new functionality?
- [ ] Or are manual tests documented (at least on this ticket)?
- [ ] Are any Playbook/Handbook pages updated?
- [ ] Has a new module release (patch/minor) been created/scheduled?
- [ ] Have the appropriate `backport` labels been added to the PR?

**Comment by mikelittle** [comment]
After some investigation, I managed to trigger a full call stack error:

```
Stack trace:
#0 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(324): Altis\Enhanced_Search\log_remote_request_errors(Array, 'index_exists')
#1 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#2 /usr/src/app/wordpress/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /usr/src/app/vendor/10up/elasticpress/includes/classes/Elasticsearch.php(1357): do_action('ep_remote_reque...', Array, 'index_exists')
#4 /usr/src/app/vendor/10up/elasticpress/includes/classes/Elasticsearch.php(1125): ElasticPress\Elasticsearch->remote_request('altisdevaltisde...', Array, Array, 'index_exists')
#5 /usr/src/app/packages/enhanced-search/inc/namespace.php(581): ElasticPress\Elasticsearch->index_exists('altisdevaltisde...')
#6 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(326): Altis\Enhanced_Search\filter_index_name('altisdevaltisde...')
#7 /usr/src/app/wordpress/wp-includes/plugin.php(205): WP_Hook->apply_filters('altisdevaltisde...', Array)
#8 /usr/src/app/vendor/10up/elasticpress/includes/classes/Indexable.php(150): apply_filters('ep_index_name', 'altisdevaltisde...', 2, Object(ElasticPress\Indexable\Post\Post))
#9 /usr/src/app/vendor/10up/elasticpress/includes/classes/Indexable.php(254): ElasticPress\Indexable->get_index_name(2)
#10 /usr/src/app/vendor/10up/elasticpress/includes/classes/Indexable/Post/SyncManager.php(736): ElasticPress\Indexable->delete_index()
#11 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(326): ElasticPress\Indexable\Post\SyncManager->action_create_blog_index(Object(WP_Site))
#12 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#13 /usr/src/app/wordpress/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#14 /usr/src/app/wordpress/wp-includes/ms-site.php(105): do_action('wp_initialize_s...', Object(WP_Site), Array)
#15 /usr/src/app/packages/core/inc/global_content/namespace.php(148): wp_insert_site(Array)
#16 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(326): Altis\Global_Content\maybe_create_site(Array)
#17 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#18 /usr/src/app/wordpress/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#19 /usr/src/app/packages/core/inc/class-command.php(40): do_action('altis.migrate', Array, Array)
#20 [internal function]: Altis\Command->migrate(Array, Array)
#21 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php(100): call_user_func(Array, Array, Array)
#22 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
#23 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(497): call_user_func(Object(Closure), Array, Array)
#24 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(441): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
#25 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(464): WP_CLI\Runner->run_command(Array, Array)
#26 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1296): WP_CLI\Runner->run_command_and_exit()
#27 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner->start()
#28 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/bootstrap.php(83): WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
#29 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php(32): WP_CLI\bootstrap()
#30 phar:///usr/local/bin/wp/php/boot-phar.php(20): include('phar:///usr/loc...')
#31 /usr/local/bin/wp(4): include('phar:///usr/loc...')
#32 {main}
  thrown in /usr/src/app/packages/enhanced-search/inc/namespace.php on line 403
```

This seems to be caused by the creation of the Global Media repo during Local Server stack set up.
The `altis.migrate` action creates the new sub-site, and a standard `ElasticPress` action `\ElasticPress\Indexable\Post\SyncManager::action_create_blog_index()` hooked to 'wp_initialize_site' calls delete index (even though it doesn't exist). Which, in turn, causes an `ElasticPress` exception.
We always log EP exceptions. But It is logged to the Docker output stream at this point, as I don't think the stack has finished setting up the captured logs.

In other words, I'm not sure that this is fixable unless we disable the EP exception logging just before running the code to create the Global Media site. That feels like too much cross module knowledge to me. 

In normal operation, it will only occur the first time you create a stack, so it should be rarely seen.

Closing as "won't fix".
